### PR TITLE
Tweak completion item selection to prefer matched case

### LINKED ISSUE #49813: IntelliSense turns properties used as arguments into lowercase
**Version Used**: VS 16.9.0 Preview 1

**Steps to Reproduce**:

1. Paste the following code:
```csharp
class Repro
{
    public int Length { get; }

    public void Test(Range x)
    {
    }
}
```
2. Try to type the following within `Test`:
```csharp
var (offset, length) = x.GetOffsetAndLength(Length);
```

**Expected Behavior**: you should be able to type that just fine.

**Actual Behavior**: IntelliSense will make `Length` lowercase in the argument, breaking the code.

**Screenshot**:

yqytdKpaL1

### PR DESCRIPTION
**Pull request by genlu**
An attempt to fix #49813
Just want to see what this change breaks

### CODE REVIEW

File: src/Features/Core/Portable/Completion/CompletionHelper.cs [hunk 1]
```diff
@@ -250,12 +250,6 @@ private int CompareMatches(PatternMatch match1, PatternMatch match2, CompletionI
                 return diff;
             }
 
-            var preselectionDiff = ComparePreselection(item1, item2);
```
Comment (genlu) [inline src/Features/Core/Portable/Completion/CompletionHelper.cs:253]: Looks like remove this check caused `VisualBasicCompletionCommandHandlerTests.BooleanPreselection2` to fail. After reading thru the comments in https://github.com/dotnet/roslyn/issues/4892, I think we need to preserve the behavior in VB that type `f` and `t` would preselect `False` and `True`.

File: src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/MatchResult.cs [hunk 2]
```diff
@@ -89,14 +89,14 @@ public int CompareTo(MatchResult other, string filterText)
             ImmutableArray.Create<Func<MatchResult, string, IComparable>>(
                 // Prefer the item that matches a longer prefix of the filter text.
                 (f, s) => f.RoslynCompletionItem.FilterText.GetCaseInsensitivePrefixLength(s),
+                // If there are "Abc" vs "abc", we should prefer the case typed by user.
+                (f, s) => f.RoslynCompletionItem.FilterText.GetCaseSensitivePrefixLength(s),
```
Comment (ryzngard) [inline src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/MatchResult.cs:93]: Can we add a test that demonstrates this change in behavior?

Comment (genlu) [reply src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/MatchResult.cs:93]: Added

**Review by allisonchou — APPROVED** [review]

**Comment by genlu** [comment]
Thanks @allisonchou!

**Pull request merged**